Update CI to use mypy <0.990

mypy update 0.990 broke the CI, causing many errors. Previous version 0.982 works

```
python3 -m mypy arkouda
arkouda/logger.py:138: error: Incompatible default for argument "handlerNames" (default has type "None", argument has type "List[str]")  [assignment]
arkouda/logger.py:138: note: PEP 484 prohibits implicit Optional. Accordingly, mypy has changed its default to no_implicit_optional=True
arkouda/logger.py:138: note: Use https://github.com/hauntsaninja/no_implicit_optional to automatically upgrade your codebase
arkouda/logger.py:245: error: Incompatible default for argument "logLevel" (default has type "None", argument has type "LogLevel")  [assignment]
arkouda/logger.py:245: note: PEP 484 prohibits implicit Optional. Accordingly, mypy has changed its default to no_implicit_optional=True
arkouda/logger.py:245: note: Use https://github.com/hauntsaninja/no_implicit_optional to automatically upgrade your codebase
arkouda/message.py:287: error: Incompatible default for argument "token" (default has type "None", argument has type "str")  [assignment]
arkouda/message.py:287: note: PEP 484 prohibits implicit Optional. Accordingly, mypy has changed its default to no_implicit_optional=True
arkouda/message.py:287: note: Use https://github.com/hauntsaninja/no_implicit_optional to automatically upgrade your codebase
arkouda/message.py:289: error: Incompatible default for argument "args" (default has type "None", argument has type "str")  [assignment]
arkouda/message.py:289: note: PEP 484 prohibits implicit Optional. Accordingly, mypy has changed its default to no_implicit_optional=True
arkouda/message.py:289: note: Use https://github.com/hauntsaninja/no_implicit_optional to automatically upgrade your codebase
arkouda/series.py:396: error: Incompatible default for argument "index_labels" (default has type "None", argument has type "List[str]")  [assignment]
arkouda/series.py:396: note: PEP 484 prohibits implicit Optional. Accordingly, mypy has changed its default to no_implicit_optional=True
arkouda/series.py:396: note: Use https://github.com/hauntsaninja/no_implicit_optional to automatically upgrade your codebase
arkouda/series.py:396: error: Incompatible default for argument "value_label" (default has type "None", argument has type "str")  [assignment]
arkouda/series.py:527: error: Incompatible default for argument "index_labels" (default has type "None", argument has type "List[str]")  [assignment]
arkouda/series.py:527: note: PEP 484 prohibits implicit Optional. Accordingly, mypy has changed its default to no_implicit_optional=True
arkouda/series.py:527: note: Use https://github.com/hauntsaninja/no_implicit_optional to automatically upgrade your codebase
arkouda/series.py:527: error: Incompatible default for argument "value_labels" (default has type "None", argument has type "List[str]")  [assignment]
arkouda/series.py:598: error: Incompatible default for argument "labels" (default has type "None", argument has type "Strings")  [assignment]
arkouda/series.py:598: note: PEP 484 prohibits implicit Optional. Accordingly, mypy has changed its default to no_implicit_optional=True
arkouda/series.py:598: note: Use https://github.com/hauntsaninja/no_implicit_optional to automatically upgrade your codebase
arkouda/pdarraycreation.py:133: error: Incompatible default for argument "dtype" (default has type "None", argument has type "Union[dtype[Any], type, str]")  [assignment]
arkouda/pdarraycreation.py:133: note: PEP 484 prohibits implicit Optional. Accordingly, mypy has changed its default to no_implicit_optional=True
arkouda/pdarraycreation.py:133: note: Use https://github.com/hauntsaninja/no_implicit_optional to automatically upgrade your codebase
arkouda/pdarraycreation.py:713: error: Incompatible default for argument "seed" (default has type "None", argument has type "Union[int, signedinteger[_8Bit], signedinteger[_16Bit], signedinteger[_32Bit], signedinteger[_64Bit], unsignedinteger[_8Bit], unsignedinteger[_16Bit], unsignedinteger[_32Bit], unsignedinteger[_64Bit]]")  [assignment]
arkouda/pdarraycreation.py:713: note: PEP 484 prohibits implicit Optional. Accordingly, mypy has changed its default to no_implicit_optional=True
arkouda/pdarraycreation.py:713: note: Use https://github.com/hauntsaninja/no_implicit_optional to automatically upgrade your codebase
arkouda/pdarrayIO.py:629: error: Incompatible default for argument "names" (default has type "None", argument has type "List[str]")  [assignment]
arkouda/pdarrayIO.py:629: note: PEP 484 prohibits implicit Optional. Accordingly, mypy has changed its default to no_implicit_optional=True
arkouda/pdarrayIO.py:629: note: Use https://github.com/hauntsaninja/no_implicit_optional to automatically upgrade your codebase
arkouda/pdarrayIO.py:719: error: Incompatible default for argument "write_file" (default has type "None", argument has type "str")  [assignment]
arkouda/pdarrayIO.py:719: note: PEP 484 prohibits implicit Optional. Accordingly, mypy has changed its default to no_implicit_optional=True
arkouda/pdarrayIO.py:719: note: Use https://github.com/hauntsaninja/no_implicit_optional to automatically upgrade your codebase
arkouda/pdarrayIO.py:796: error: Incompatible default for argument "write_file" (default has type "None", argument has type "str")  [assignment]
arkouda/pdarrayIO.py:796: note: PEP 484 prohibits implicit Optional. Accordingly, mypy has changed its default to no_implicit_optional=True
arkouda/pdarrayIO.py:796: note: Use https://github.com/hauntsaninja/no_implicit_optional to automatically upgrade your codebase
arkouda/join.py:170: error: Incompatible default for argument "wherefunc" (default has type "None", argument has type "Callable[..., Any]")  [assignment]
arkouda/join.py:170: note: PEP 484 prohibits implicit Optional. Accordingly, mypy has changed its default to no_implicit_optional=True
arkouda/join.py:170: note: Use https://github.com/hauntsaninja/no_implicit_optional to automatically upgrade your codebase
arkouda/join.py:170: error: Incompatible default for argument "whereargs" (default has type "None", argument has type "Tuple[pdarray, pdarray]")  [assignment]
arkouda/groupbyclass.py:1422: error: Incompatible default for argument "user_defined_name" (default has type "None", argument has type "str")  [assignment]
arkouda/groupbyclass.py:1422: note: PEP 484 prohibits implicit Optional. Accordingly, mypy has changed its default to no_implicit_optional=True
arkouda/groupbyclass.py:1422: note: Use https://github.com/hauntsaninja/no_implicit_optional to automatically upgrade your codebase
arkouda/dataframe.py:289: note: By default the bodies of untyped functions are not checked, consider using --check-untyped-defs  [annotation-unchecked]
arkouda/client.py:107: error: Incompatible default for argument "access_token" (default has type "None", argument has type "str")  [assignment]
arkouda/client.py:107: note: PEP 484 prohibits implicit Optional. Accordingly, mypy has changed its default to no_implicit_optional=True
arkouda/client.py:107: note: Use https://github.com/hauntsaninja/no_implicit_optional to automatically upgrade your codebase
arkouda/client.py:157: error: Incompatible types in assignment (expression has type "Optional[str]", variable has type "str")  [assignment]
arkouda/client.py:365: error: Incompatible default for argument "args" (default has type "None", argument has type "str")  [assignment]
arkouda/client.py:365: note: PEP 484 prohibits implicit Optional. Accordingly, mypy has changed its default to no_implicit_optional=True
arkouda/client.py:365: note: Use https://github.com/hauntsaninja/no_implicit_optional to automatically upgrade your codebase
arkouda/client.py:435: error: Incompatible default for argument "args" (default has type "None", argument has type "str")  [assignment]
arkouda/client.py:435: note: PEP 484 prohibits implicit Optional. Accordingly, mypy has changed its default to no_implicit_optional=True
arkouda/client.py:435: note: Use https://github.com/hauntsaninja/no_implicit_optional to automatically upgrade your codebase
arkouda/client.py:573: error: Incompatible default for argument "json_obj" (default has type "None", argument has type "Dict[Any, Any]")  [assignment]
arkouda/client.py:573: note: PEP 484 prohibits implicit Optional. Accordingly, mypy has changed its default to no_implicit_optional=True
arkouda/client.py:573: note: Use https://github.com/hauntsaninja/no_implicit_optional to automatically upgrade your codebase
arkouda/client.py:614: error: Incompatible default for argument "args" (default has type "None", argument has type "Dict[Any, Any]")  [assignment]
arkouda/client.py:614: note: PEP 484 prohibits implicit Optional. Accordingly, mypy has changed its default to no_implicit_optional=True
arkouda/client.py:614: note: Use https://github.com/hauntsaninja/no_implicit_optional to automatically upgrade your codebase
arkouda/client.py:615: error: Incompatible default for argument "payload" (default has type "None", argument has type "memoryview")  [assignment]
arkouda/client.py:615: note: PEP 484 prohibits implicit Optional. Accordingly, mypy has changed its default to no_implicit_optional=True
arkouda/client.py:615: note: Use https://github.com/hauntsaninja/no_implicit_optional to automatically upgrade your codebase
Found 24 errors in 8 files (checked 35 source files)
make: *** [Makefile:428: mypy] Error 1
```

--- a/REQUIREMENTS.md
+++ b/REQUIREMENTS.md
@@ -38,7 +38,7 @@ The dependencies listed here are only required if you will be doing development 
 - `sphinx-argparse`
 - `sphinx-autoapi`
 - `typed-ast`
-- `mypy>=0.931`
+- `mypy>=0.931,<0.990`
 - `flake8`
 
 ## Installing/Updating Python Dependencies

--- a/arkouda-env-dev.yml
+++ b/arkouda-env-dev.yml
@@ -26,7 +26,7 @@ dependencies:
   - sphinx-autoapi
   - typed-ast
   - flake8
-  - mypy>=0.931
+  - mypy>=0.931,<0.990
   - black
   - isort
   

--- a/pydoc/requirements.txt
+++ b/pydoc/requirements.txt
@@ -22,5 +22,5 @@ Sphinx>=5.1.1
 sphinx-argparse
 sphinx-autoapi
 typed-ast
-mypy>=0.931
+mypy>=0.931,0.990
 flake8

--- a/setup.py
+++ b/setup.py
@@ -157,7 +157,7 @@ setup(
     extras_require={  # Optional
         'dev': ['pexpect', 'pytest>=6.0', 'pytest-env',
                 'Sphinx>=5.1.1', 'sphinx-argparse', 'sphinx-autoapi',
-                'mypy>=0.931', 'typed-ast', 'black', 'isort',
+                'mypy>=0.931,<0.990', 'typed-ast', 'black', 'isort',
                 'flake8'],
     },
     # replace original install command with version that also builds

